source/client-side-encryption/tests/README.rst

```diff
@@ -2012,3 +2012,72 @@ the environment.
 .. _Automatic AWS Credentials: ../client-side-encryption.rst#automatic-aws-credentials
 .. _ClientEncryption: ../client-side-encryption.rst#clientencryption
 .. _auth-aws: ../../auth/auth.rst#obtaining-credentials
+
+16. On-demand GCP Credentials
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Refer: `Automatic GCP Credentials`_.
+
+For these cases, create a ClientEncryption_ object :math:`C` with the following
+options:
+
+.. code-block:: typescript
+
+   ClientEncryptionOpts {
+      keyVaultClient: <setupClient>,
+      keyVaultNamespace: "keyvault.datakeys",
+      kmsProviders: { "gcp": {} },
+   }
+
+Case 1: Failure
+```````````````
+
+Do not run this test case in an environment with a GCP service account is
+attached (e.g. any `GCE equivalent runtime
+<https://google.aip.dev/auth/4115>`_). This may be run in an AWS EC2 instance.
+
+Attempt to create a datakey with :math:`C` using the ``"gcp"`` KMS provider and
+following ``DataKeyOpts``:
+
+.. code-block:: typescript
+
+   class DataKeyOpts {
+      masterKey: {
+         "projectId": "devprod-drivers",
+         "location": "global",
+         "keyRing": "key-ring-csfle",
+         "keyName": "key-name-csfle"
+      }
+   }
+
+Expect the attempt to obtain ``"gcp"`` credentials from the environment to fail.
+
+Case 2: Success
+```````````````
+
+This test case must run in a Google Compute Engine (GCE) Virtual Machine with a
+service account attached. See `drivers-evergreen-tools/.evergreen/csfle/gcpkms
+<https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/csfle/gcpkms>`_
```

Comment:
Flagging that this link will resolve once https://github.com/mongodb-labs/drivers-evergreen-tools/pull/216 is merged.